agot-bg-game-server/src/common/ingame-game-state/game-data-structure/Game.ts

```diff
@@ -148,6 +149,34 @@ export default class Game {
         return unit;
     }
 
+    transformSingle(unit: Unit, targetType: UnitType): Unit {
```

Comment:
I would expect this method to also handle client-syncing by sending messages. Either put the code inside it inside `transformUnits`.

I gues that you did it that way because in `PlayerMusteringGameState` where you call it, you don't want to send messages, but I think it's better to change PlayerMustering then.

Reply:
Yeh, that's exactly why its like that. I originally had only `transformUnits` but when I saw `PlayerMusteringGameState` handles messaging differently I thought splitting it to make the change as seamless as possible is better. 
So, to go on with this change, I should move all the logic back into one method that handles messaging + unit creation. Then remove the `add-units` and `remove-units` messages later in `PlayerMusteringGameState`, right?

Reply:
Further looking into this, if I don't keep a version that doesn't handle the transformation without message creation, it would resort in excessive messages between the server and clients as up to 2 messages will be sent per region (1 for new units, 1 for upgraded units), this is unavoidable so long as the messages are sent within `transformUnits`. This might be a non-issue since it can happen only in Strongholds when a player both upgrades and musters a new land unit. Regardless, I wait to hear your opinions before acting on this.

Reply:
>So, to go on with this change, I should move all the logic back into one method that handles messaging + unit creation. Then remove the add-units and remove-units messages later in PlayerMusteringGameState, right?

Yup!

In general, for this kind of trade-off, code readability always outweights performance. It'd be better to have one message sent with the whole mustering for all regions, but I feel like it would make the code readable to have the messages put there.

Reply:
Okay, I did it